Compile error

Current master does not compile in my enviroment (g++ (Debian 10.2.1-6) 10.2.1 20210110)

```
[  0%] Building CXX object CMakeFiles/freeorioncommon.dir/util/StringTable.cpp.o
(lots of warnings skipped)
...
util/StringTable.cpp:312:48: error: no match for ‘operator|’ (operand types are ‘boost::unordered::unordered_map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, StringTable::hasher, StringTable::equalizer>’ and ‘const std::ranges::views::__adaptor::_RangeAdaptorClosure<<lambda(_Range&&)> >’)
  312 |         for (auto& user_read_entry : m_strings | range_values) {
      |                                      ~~~~~~~~~ ^ ~~~~~~~~~~~~
      |                                      |           |
      |                                      |           const std::ranges::views::__adaptor::_RangeAdaptorClosure<<lambda(_Range&&)> >
      |                                      boost::unordered::unordered_map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, StringTable::hasher, StringTable::equalizer>
In file included from /usr/include/c++/10/bits/ranges_algobase.h:38,
                 from /usr/include/c++/10/bits/ranges_uninitialized.h:36,
                 from /usr/include/c++/10/memory:69,
                 from /usr/include/boost/container_hash/extensions.hpp:35,
                 from /usr/include/boost/container_hash/hash.hpp:761,
                 from /usr/include/boost/functional/hash.hpp:6,
                 from /usr/include/boost/unordered/unordered_map.hpp:18,
                 from /usr/include/boost/unordered_map.hpp:17,
                 from /home/muzz/freeorion-project/Grummel7/util/StringTable.h:7,
                 from /home/muzz/freeorion-project/Grummel7/util/StringTable.cpp:1:
```

--- a/util/ranges.h
+++ b/util/ranges.h
@@ -1,7 +1,8 @@
 #ifndef _Range_Fallback_Include_
 #define _Range_Fallback_Include_
 
-#if __has_include(<ranges>) && (!defined(__clang_major__) || (__clang_major__ >= 16))
+#if __has_include(<ranges>) && (!defined(__clang_major__) || (__clang_major__ >= 16)) \
+    && (!defined(__GNUC__) || (__GNUC__ >= 11) || (__GNUC__ == 10 && __GNUC_MINOR__ >= 4))
 # include <ranges>
 static constexpr auto& range_keys = std::views::keys;
 static constexpr auto& range_values = std::views::values;

